lib/Domain/Refactor/GenerateAccessor.php

```diff
@@ -2,9 +2,24 @@
 
 namespace Phpactor\CodeTransform\Domain\Refactor;
 
+use InvalidArgumentException;
 use Phpactor\CodeTransform\Domain\SourceCode;
+use RuntimeException;
 
 interface GenerateAccessor
 {
-    public function generateAccessor(SourceCode $sourceCode, int $offset): SourceCode;
+    /**
+     * Generates an accessor.
+     *
+     * @param SourceCode $sourceCode The source code of the working file.
+     * @param string $propertyName The name of the property to generate the accessor for.
+     * @param int $offset The position of the cursor, used to identified the class if there
+     * is more than one in the `$sourceCode`.
+     *
+     * @return SourceCode
+     *
+     * @throws InvalidArgumentException If there is no class in the code.
+     * @throws RuntimeException If it's impossible to determine which class to use.
+     */
+    public function generate(SourceCode $sourceCode, string $propertyName, int $offset): SourceCode;
```

Comment:
Again, here I'd be happy (if not particularly fussed) about the exception docs, but the rest is self-evident?

Reply:
As above, as was just worried about confusion since in every other places I looked the offset is used to identify the generated item.
In this case it's used to identify the class and not the property, if you are cool without the doc I don't mind.